test/cql-pytest/test_describe.py

```diff
@@ -2567,3 +2568,90 @@ def aux(row: DescRow) -> DescRow:
             assert restore_stmts == res
         finally:
             cql.execute(f"DROP KEYSPACE IF EXISTS {ks}")
+
+### ===========================================================================
+
+# `varchar` is an alias for `text`. Scylla defaults to `text` and we test it here.
+# It's not very important, but it's better to have a test that detects changes
+# like that and verifies we handle that case as expected.
+def test_restore_udf_with_varchar(cql, test_keyspace):
+    fn_content = """\
+        (arg {type_name})
+        RETURNS NULL ON NULL INPUT
+        RETURNS {type_name}
+        LANGUAGE lua
+        AS $$
+        return arg
+        $$;"""
+    # Get rid of indentation in the string.
+    fn_content = textwrap.dedent(fn_content)
+
+    with new_function(cql, test_keyspace, fn_content.format(type_name="varchar")) as fn:
+        udf_name = fn
+        udf_row = cql.execute(f"DESC FUNCTION {test_keyspace}.{fn}").one()
+        udf_stmt = udf_row.create_statement
+
+        expected_stmt = f"CREATE FUNCTION {test_keyspace}.{fn}{fn_content.format(type_name="text")}"
+        assert udf_stmt == expected_stmt
+
+    try:
+        cql.execute(udf_stmt)
+    finally:
+        cql.execute(f"DROP FUNCTION IF EXISTS {test_keyspace}.{udf_name}")
+
+# Verifies that restoring a UDF with built-in types works correctly.
```

Comment:
Again, I don't think you need to use the word "restore" (or actually execute the statement you get from DESC FUNCTION.
And since you use lua, I guess this test is Scylla-only.